src/SIL.Harmony.Sample/Changes/SetTagChange.cs

```diff
@@ -8,19 +8,26 @@ public class SetTagChange(Guid entityId, string text) : Change<Tag>(entityId), I
 {
     public string Text { get; } = text;
 
-    public override ValueTask<Tag> NewEntity(Commit commit, IChangeContext context)
+    public override async ValueTask<Tag> NewEntity(Commit commit, IChangeContext context)
     {
-        return new(new Tag()
+        var tagExists = await context.GetObjectsOfType<Tag>(nameof(Tag)).AnyAsync(t => t.Text == Text);
```

Comment:
It's a bit dissatisfying that we have to tell it the name of the discriminator, because we already tell Harmony that when it persists the snapshots.

Reply:
yeah I considered making that better, however when using an object adapter (like we are in lcm) it actually needs to use that adapter, so I just decided that the project can make a helper method to figure out the type on it's own.